relay-server/src/extractors/received_at.rs

```diff
@@ -0,0 +1,65 @@
+use std::convert::Infallible;
+
+use axum::extract::FromRequestParts;
+use axum::http::request::Parts;
+use axum::Extension;
+use chrono::{DateTime, Utc};
+
+/// The time at which the request started.
+#[derive(Clone, Copy, Debug)]
+pub struct ReceivedAt(DateTime<Utc>);
```

Comment:
```suggestion
pub struct ReceivedAt(pub DateTime<Utc>);
```
No point in having it private, public also means you can destructure it on assignment `let ReceivedAt(received_at) = envelope.received_at()`.